src/test/java/org/minions/devfund/sergio/WhackAMoleTest.java

```diff
@@ -0,0 +1,147 @@
+        this.whackAMole = new WhackAMole(ATTEMPTS, GRID_DIMENSION, TOTAL_MOLES);
+    }
+
+    /**
+     * Verifies if the place method works fine.
+     */
+    @Test
+    public void verifyPlaceAMoleWorks() {
+        final int posX = 1;
+        final int posY = 5;
+        assertTrue(whackAMole.place(posX, posY));
+    }
+
+    /**
+     * Verifies if it not possible place a mole in a mole already created.
+     */
+    @Test
+    public void verifyIsAlreadyAMole() {
+        final int posX = 1;
+        final int posY = 1;
+        whackAMole.place(posX, posY);
+        assertFalse(whackAMole.place(posX, posY));
+    }
+
+    /**
+     * Verifies if whack works in the mole position.
+     */
+    @Test
+    public void verifyWhackAMoleWorks() {
+        final int posX = 3;
+        final int posY = 3;
+        whackAMole.place(posX, posY);
+        whackAMole.whack(posX,  posY);
+        assertEquals('W', whackAMole.getStatusPosition(posX,  posY));
+    }
+
+    /**
+     * Verifies if whack miss is no change of grid.
+     */
+    @Test
+    public void verifyWhackMissWorks() {
+        final int posX = 2;
+        final int posY = 1;
+        whackAMole.whack(posX, posY);
+        assertEquals('*', whackAMole.getStatusPosition(posX, posY));
+    }
+
+    /**
+     * Verifies that the number of attempts id changed after a try.
+     */
+    @Test
+    public void verifyAttemptsIsChangeAfterAWhack() {
+        final int posX = 4;
+        final int posY = 2;
+        whackAMole.whack(posX, posY);
+        assertEquals(ATTEMPTS - 1, whackAMole.getAttemptsLeft());
+    }
+
+    /**
+     * Verifies that the score is changed after a whack hit.
+     */
+    @Test
+    public void verifyScoreIsWorking() {
+        final int posX = 1;
+        final int posY = 2;
+        whackAMole.place(posX, posY);
+        whackAMole.whack(posX,  posY);
+        assertEquals(SCORE + 1, whackAMole.getScore());
+    }
+
+    /**
+     * Verifies if the grid for user no show the moles.
+     */
+    @Test
+    public void verifyUserGridShowCorrectly() {
+        final int posX = 4;
+        final int posY = 5;
+        whackAMole.place(posX, posY);
+        StringBuilder test = new StringBuilder();
+        for (int i = 0; i < GRID_DIMENSION; i++) {
+            for (int j = 0; j < GRID_DIMENSION; j++) {
+                test.append('*');
+            }
+            test.append("\n");
+        }
+        assertEquals(test.toString().trim(), whackAMole.getUserGrid().toString().trim());
+    }
+
+    /**
+     * Verifies that get grid is working.
+     */
+    @Test
+    public void verifyGetGridIsWorking() {
+        final int posX = 2;
+        final int posY = 2;
+        whackAMole.place(posX, posY);
+        StringBuilder test = new StringBuilder();
+        for (int i = 0; i < GRID_DIMENSION; i++) {
+            for (int j = 0; j < GRID_DIMENSION; j++) {
+                boolean positionFound = i == posX && j == posY;
+                if (positionFound) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)